### PR TITLE
schdule: exclude offline store in filter source

### PR DESCRIPTION
**Pull request by nolouch**
fix https://github.com/pingcap/pd/issues/974

### CODE REVIEW

**Comment by disksing** [comment]
How about `return store.IsTombstone()` directly? BTW, please add some test.

File: server/schedulers/balance_test.go [hunk 1]
```diff
@@ -228,6 +228,10 @@ func (s *testBalanceLeaderSchedulerSuite) TestBalanceFilter(c *C) {
 
 	CheckTransferLeader(c, s.schedule(nil), schedule.OpBalance, 4, 1)
 	// Test stateFilter.
+	// if store 4 if offline, we schould consider it
+	// because it still providers services
```
Comment (disksing) [inline server/schedulers/balance_test.go]: providers -> provides.

**Comment by disksing** [comment]
PTAL @huachaohuang

File: server/schedule/filters.go [hunk 2]
```diff
@@ -100,7 +100,7 @@ func (f *stateFilter) filter(store *core.StoreInfo) bool {
 }
 
 func (f *stateFilter) FilterSource(opt Options, store *core.StoreInfo) bool {
-	return f.filter(store)
+	return store.IsTombstone()
 }
 
 func (f *stateFilter) FilterTarget(opt Options, store *core.StoreInfo) bool {
```
Comment (huachaohuang) [inline server/schedule/filters.go]: You can move `filter()` into here.

**Pull request merged**